Curl build improvements

tkelman opened the pull request:
Should fix the from-scratch build of master which was broken by #18103

tkelman committed:
Delete some files from libssh2 and curl that keep changing on travis

causing the cache to get re-uploaded

tkelman committed:
Make rpath settings a bit more consistent

probably addresses #18106

tkelman committed:
Install all 3 mbedtls dlls on windows

and build tests

tkelman committed:
disable libpsl in curl build

should fix clean build of master

indent line continuations

tkelman committed:
disable using zlib in mbedtls, libssh2, and libcurl

since it requires additional dev headers
fix #18101

tkelman committed:
Cleanup: replace misleading STAGE#_DEPS variables with DEP_LIBS

vtjnash commented on deps/mbedtls.mk:
```diff
@@ -56,8 +52,10 @@ endif
 
 $(MBEDTLS_OBJ_TARGET): $(MBEDTLS_OBJ_SOURCE) | $(build_shlibdir)
-ifeq ($(OS), WINNT)
+ifeq ($(BUILD_OS), WINNT)
```
this line diff is wrong. the install target should never depend on BUILD_OS

tkelman replied:
staged install doesn't work properly on msys2

tkelman replied:
I'll put it back for now though since it's putting .dll.a files in the wrong place otherwise, need to fix that upstream. I plan on moving appveyor over to a cygwin-cross and dropping official support for MSYS2 in the future since it's not actively maintained and is the cause of too many of these special cases.

tkelman committed:
Improve handling of curl, mbedtls, libssh2 USE_SYSTEM

fix #17884

tkelman committed:
Set CMAKE_INSTALL_LIBDIR and CMAKE_INSTALL_BINDIR

in CMAKE_COMMON, so libssh2 installs without an x86_64-linux-gnu
multiarch prefix

tkelman committed:
Disable building static mbedtls library

since we don't need it

Go back to just copying the mbedtls dll even in cygwin-cross,
otherwise it puts .dll.a files that we don't need alongside the dlls
(should probably fix this upstream)

tkelman committed:
Also just copy libgit2 dll for its install target

since otherwise it puts pkg-config files in LIB_INSTALL_DIR
which we don't need - if mbedtls separates runtime from library
destinations then we could separate BIN_INSTALL_DIR from LIB_INSTALL_DIR
or convince libgit2 to use the standard GNUInstallDirs names

tkelman commented:
A todo for myself, can wait until a separate PR though - libgit2's finding of curl via pkg-config should be adjusted with a patch to either fail loudly if pkg-config isn't found and the library would otherwise be built without proxy support, or avoid using pkg-config to find libcurl.

tkelman commented:
merging so nanosoldier works on tonight's daily build
